remix-analyzer/src/solidity-analyzer/modules/gasCosts.ts

```diff
@@ -60,4 +55,18 @@ export default class gasCosts {
     })
     return report
   }
+
+  /**
+  * call the given @arg cb (function) for all the contracts. Uses last compilation result
+  * stop visiting when cb return true
+  * @param {Function} cb    - callback
+  */
+  // @TODO has been copied from remix-ide repo ! should fix that soon !
```

Comment:
Maybe create an issue for it & reference the id of the issue here.

Reply:
this TODO is there before this PR. not sure what to be fixed :|

Reply:
XD @yann300 any idea about that ?

Reply:
I think this message refers to the double loop `for ( for (`
but the Blame github function does'nt really help finding the reason..

Reply:
btw looks like this stop visiting when a `report` is returned. Is that right? cause that way it seems some other contracts will not be checked at all...

Reply:
but in the devdoc, it is written like that is an expected behaviour. Not sure if we need to create an issue to know what is the issue 🤔

Reply:
Yeah probably not in this PR, but we need to understand why it's being done like that...

Reply:
created https://github.com/ethereum/remix/issues/1435 to track this